build-scripts/rebuild_libs.rs

```diff
@@ -0,0 +1,82 @@
+/*
+* This is a script to clobber the {desktop, ios, android} directories
+* and rebuild libs via the build-all.sh script.
+*/
+use std::fs;
+use std::path::Path;
+use std::env;
+use std::process::Command;
+use std::io::prelude::*;
+
+#[allow(unused_must_use)]
+fn check_version() -> bool {
+
+    let mut clobber_needed: bool = false;
+    let root = Path::new("../libs");
+
+    assert!(env::set_current_dir(&root).is_ok());
+
+    let mut file = fs::File::open(".local_libs_version").expect("Counld not rebuild libs. Delete your folders for desktop, ios, and android, then manually run libs/build-all.sh \n");
+    let mut local_version = String::new();
+    file.read_to_string(&mut local_version);
+
+    //do I need to do further file parsing?
+    let mut file2 = fs::File::open("version_log").expect("Counld not rebuild libs. Delete your folders for desktop, ios, and android, then manually run libs/build-all.sh \n");
+    let mut newest_version = String::new();
+    file2.read_to_string(&mut newest_version);
+
+    //this reads the most recent version number. (everything before the '.')
+    let mut s = "".to_owned();
+    for c in newest_version.chars() {
+        if c == '.' {
+            break;
+        } else {
+            s = s + &c.to_string();
+        }
+    }
+    let mut t = "".to_owned();
+    for c in local_version.chars() {
+        if c == '.' {
+            break;
+        } else {
+            t = t + &c.to_string();
+        }
+    }
```

Comment:
I'm not comfortable with file IO and this seems really messy, but it works.

Reply:
It is a little messy. :)

It is also fragile to the order of things in the file, the line format, etc. Consider this version, which is robust to this stuff. It needs a comment or two and  better variable names, though.

https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=b39051e9855162cf5a643037da8207ec

Reply:
Thank you for writing this, I was really struggling